src/sketch.js

```diff
@@ -212,8 +223,31 @@ function draw()
     // set background color
     background(...BACKGROUND_COLOR);
 
+    const step_params = {x: 0.2, y: 0.2, speed: 0.18}
+
+    if(soundOn)
+    {
+        // vary terrain speed based on sound amplitude
+        step_params.speed = map(amplitude.getLevel(), 0, 1, AUDIO_TO_SPEED.min, AUDIO_TO_SPEED.max);
+
+        // change the sound pan
+        alterPan();
+
+        let a = fft.analyze();
+
+        BACKGROUND_COLOR[0] = (a[0] + a[1]) / 2;
+        BACKGROUND_COLOR[1] = (a[2] + a[3]) / 2;
+        BACKGROUND_COLOR[2] = (a[4] + a[5]) / 2;
+        TERRAIN_COLOR[0] = (a[6] + a[7] + a[8]) / 5;
+        TERRAIN_COLOR[1] = (a[9] + a[10] + a[11]) / 4;
+        TERRAIN_COLOR[2] = (a[12] + a[13] + a[14]) / 4;
+        TERRAIN_COLOR[3] = Math.max(a[15], 80);
```

Comment:
80 should be a configuration constant?

Reply:
This might be in a future implementation. For now it's very uncertain how this really behaves. 😕